client/src/components/signup.js

```diff
@@ -0,0 +1,269 @@
+import React from "react";
+import axios from "axios";
+
+class Register extends React.Component {
+  state = {
+    username: "",
+    firstName: "",
+    lastName: "",
+    email: "",
+    gender: null,
+    birthday: null,
+    month: "January",
+    day: 1,
+    year: 1920,
+    password: "",
+    days: [],
+    years: [],
+    errorHeader: ""
+  };
+  componentWillMount() {
```

Comment:
This is awesome - the most detailed registration I've seen! I think I managed to register without checking off terms & conditions but that is far beyond the scope of this project. This is really well done and reminds me of all the information companies collect when you want to use their website.